### PR TITLE
Herwig7Interface backport 9_3_X

### PR DESCRIPTION
**Pull request by Andrej-CMS**
Dear all,
this is the backport of the H7Interface to CMSSW 93X. It superseeds Herwig++.
Kind regards,
Andrej

### CODE REVIEW

**Comment by Andrej-CMS** [comment]
to be tested with https://github.com/cms-sw/cmsdist/pull/4578

**Comment by efeyazgan** [comment]
please test with cms-sw/cmsdist#4578

**Comment by Andrej-CMS** [comment]
Dear all,
is this a real error or is this an issue with jenkins? Summary page doesn't exist

**Comment by smuzaffar** [comment]
please test with cms-sw/cmsdist#4578

**Comment by efeyazgan** [comment]
please test with cms-sw/cmsdist#4578

**Comment by efeyazgan** [comment]
+1

**Comment by kpedro88** [comment]
@Andrej-CMS is this a backport of a specific PR? please provide the original PR number

**Comment by alberto-sanchez** [comment]
backport #21711

File: GeneratorInterface/Herwig7Interface/src/Herwig7Interface.cc [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+        else
+		    choice = runModeTemp.substr(0, pos);
+        
+		if (pos == std::string::npos)
+			runModeTemp.erase();
+        else
+            runModeTemp.erase(0, pos+1);
+
+		// construct HerwigUIProvider object and return it as global object
+		HwUI_ = new Herwig::HerwigUIProvider(pset, dumpConfig_, Herwig::RunMode::READ);
```
Comment (kpedro88) [inline GeneratorInterface/Herwig7Interface/src/Herwig7Interface.cc]: This looks like a memory leak.

Comment (Andrej-CMS) [reply GeneratorInterface/Herwig7Interface/src/Herwig7Interface.cc]: I added a delete option for HwUI_ object in case it already exists

**Comment by fabiocos** [comment]
@alberto-sanchez @Andrej-CMS @efeyazgan @qliphy this version drops ThePegInterface in favour of Herwig7Interface. Are they fully incompatible? As already discussed, this means fropping support for herwigpp for herwig7, but there are 9_3_X campaigns in which herwigpp has been used to produce samples, for instance 

https://cms-pdmv.cern.ch/mcm/requests?page=0&member_of_campaign=RunIIFall17GS&generators=herwigpp

Strictly speaking this move breaks backward compatibility, it is not just the addition of a generator. Unless we decide to consider herwig7 as the update of herwigpp (in a sense it is), and whatever new sample has to be produced with this version.

**Comment by efeyazgan** [comment]
@fabiocos we discussed in PC to move to herwig7 and drop herwig++ after some certain version of 9_3_X and 7_1_X however we are waiting for jet/met to check the first herwig7 request. Once they give us green light, we can move forward.

**Comment by fabiocos** [comment]
@efeyazgan thanks for the clarification, so I think we should continue with the usual code review and integration

**Comment by efeyazgan** [comment]
@fabiocos we have the green light from jetmet. So, we can move forward.

**Comment by kpedro88** [comment]
please test with cms-sw/cmsdist#4578

**Comment by kpedro88** [comment]
@alberto-sanchez, @qliphy, @perrozzi, @efeyazgan please sign again (then we can go ahead with the new release)

**Comment by davidlange6** [comment]
Should this not also include open loop improvements to make production significantly more efficient?




On 8 Feb 2019, at 22:13, Kevin Pedro <notifications@github.com<mailto:notifications@github.com>> wrote:


@alberto-sanchez<https://github.com/alberto-sanchez>, @qliphy<https://github.com/qliphy>, @perrozzi<https://github.com/perrozzi>, @efeyazgan<https://github.com/efeyazgan> please sign again (then we can go ahead with the new release)

**Comment by kpedro88** [comment]
@davidlange6 remind me, are the open loop improvements purely technical (no change in physics)?

**Comment by davidlange6** [comment]
yes technical - saving 1/3 or 1/2 of the CPU time in a 8 hour job I guess

**Comment by kpedro88** [comment]
@davidlange I created #4688 as a combined backport of your openloops optimizations

**Comment by kpedro88** [comment]
@alberto-sanchez, @qliphy, @perrozzi, @efeyazgan ping

**Comment by qliphy** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Pull request merged**